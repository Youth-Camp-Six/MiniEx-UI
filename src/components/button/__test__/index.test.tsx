/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Button, ButtonProps } from '..';

const defaultProps = {
  onClick: jest.fn(),
};
const testProps: ButtonProps = {
  btnType: 'primary',
  size: 'lg',
  className: 'test',
};

const disabledProps: ButtonProps = {
  disabled: true,
  onClick: jest.fn(),
};

describe('test button comonent', () => {
  // 快照测试
  it('button snapshot test', () => {
    const wrapper = render(
      <div>
        <Button btnType='default'>default按钮</Button>
        <Button btnType='primary'>primary按钮</Button>
        <Button btnType='danger'>danger按钮</Button>
        <Button btnType='link'>link按钮</Button>
        <Button size='lg'>lg按钮</Button>
        <Button size='sm'>sm按钮</Button>
        <Button disabled>禁用按钮</Button>
      </div>
    );
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it('default button', () => {
    const wrapper = render(<Button {...defaultProps}>nice</Button>);
    const element = wrapper.getByText('nice') as HTMLButtonElement;
    expect(element).toBeInTheDocument();
    expect(element.tagName).toEqual('BUTTON');
    expect(element).toHaveClass('btn btn-default');
    expect(element.disabled).toBeFalsy();
    fireEvent.click(element);
    expect(defaultProps.onClick).toHaveBeenCalled();
  });

  it('different props', () => {
    const wrapper = render(<Button {...testProps}>nice</Button>);
    const element = wrapper.getByText('nice');
    expect(element).toBeInTheDocument();
    expect(element).toHaveClass('btn-primary btn-lg test');
  });

  it('disabled set to true', () => {
    const wrapper = render(<Button {...disabledProps}>Nice</Button>);
    const element = wrapper.getByText('Nice') as HTMLButtonElement;
    expect(element).toBeInTheDocument();
    expect(element.disabled).toBeTruthy();
    fireEvent.click(element);
    expect(disabledProps.onClick).not.toHaveBeenCalled();
  });
});
