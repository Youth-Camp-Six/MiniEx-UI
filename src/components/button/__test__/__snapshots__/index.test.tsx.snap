// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text default button 1`] = `
<div>
  <button
    class="btn btn-default"
  >
    default按钮
  </button>
  <button
    class="btn btn-primary"
  >
    primary按钮
  </button>
  <button
    class="btn btn-danger"
  >
    danger按钮
  </button>
  <button
    class="btn btn-link"
  >
    link按钮
  </button>
  <button
    class="btn btn-default btn-lg"
  >
    lg按钮
  </button>
  <button
    class="btn btn-default btn-sm"
  >
    sm按钮
  </button>
  <button
    class="btn btn-default disabled"
    disabled=""
  >
    禁用按钮
  </button>
</div>
`;

exports[`test button comonent button snapshot test 1`] = `
<div>
  <button
    class="btn btn-default"
  >
    default按钮
  </button>
  <button
    class="btn btn-primary"
  >
    primary按钮
  </button>
  <button
    class="btn btn-danger"
  >
    danger按钮
  </button>
  <button
    class="btn btn-link"
  >
    link按钮
  </button>
  <button
    class="btn btn-default btn-lg"
  >
    lg按钮
  </button>
  <button
    class="btn btn-default btn-sm"
  >
    sm按钮
  </button>
  <button
    class="btn btn-default disabled"
    disabled=""
  >
    禁用按钮
  </button>
</div>
`;
